src/request/Sas9RequestClient.ts

```diff
@@ -0,0 +1,117 @@
+import { AxiosRequestConfig } from 'axios'
+import axiosCookieJarSupport from 'axios-cookiejar-support'
+import * as tough from 'tough-cookie'
+import { prefixMessage } from '@sasjs/utils/error'
+import { RequestClient, throwIfError } from './RequestClient'
+
+export class Sas9RequestClient extends RequestClient {
+  constructor(baseUrl: string, allowInsecure = false) {
+    super(baseUrl, allowInsecure)
+    this.httpClient.defaults.maxRedirects = 0
+    this.httpClient.defaults.validateStatus = (status) =>
+      status >= 200 && status < 303
+
+    if (axiosCookieJarSupport) {
+      axiosCookieJarSupport(this.httpClient)
+      this.httpClient.defaults.jar = new tough.CookieJar()
+    }
+  }
+
+  public async login(username: string, password: string) {
+    const codeInjectorPath = `/User Folders/${username}/My Folder/sasjs/runner`
+    if (this.httpClient.defaults.jar) {
+      ;(this.httpClient.defaults.jar as tough.CookieJar).removeAllCookies()
+      await this.get(
+        `/SASStoredProcess/do?_program=${codeInjectorPath}&_username=${username}&_password=${password}`,
```

Comment:
The `/SASStoredProcess` part is generally fixed in 99.5% of cases but for some sites they do use a different endpoint.  It would be safer to use the `sas9path` from the sasjs config

Reply:
We now use the `jobsPath` from the config.